src/GeniusYield/TxBuilder/Class.hs

```diff
@@ -253,13 +399,92 @@ instance GYTxQueryMonad m => GYTxQueryMonad (ReaderT env m) where
     slotOfCurrentBlock = lift slotOfCurrentBlock
     logMsg ns s = withFrozenCallStack $ lift . logMsg ns s
 
-instance GYTxMonad m => GYTxMonad (ReaderT g m) where
+instance GYTxUserQueryMonad m => GYTxUserQueryMonad (ReaderT env m) where
+    ownAddresses = lift ownAddresses
+    ownChangeAddress = lift ownChangeAddress
+    ownCollateral = lift ownCollateral
+    availableUTxOs = lift availableUTxOs
+    someUTxO = lift . someUTxO
+
+instance GYTxSpecialQueryMonad m => GYTxSpecialQueryMonad (ReaderT env m) where
+    systemStart = lift systemStart
+    eraHistory = lift eraHistory
+    protocolParams = lift protocolParams
+    stakePools = lift stakePools
+
+instance GYTxMonad m => GYTxMonad (ReaderT env m) where
+    signTxBody = lift . signTxBody
+    signTxBodyWithStake = lift . signTxBodyWithStake
+    submitTx = lift . submitTx
+    awaitTxConfirmed' p = lift . awaitTxConfirmed' p
+
+-------------------------------------------------------------------------------
+-- Instances for less useful transformers, provided for completeness.
+-- Many of these transformers are fundamentally riddled with issues.
+--    See: https://github.com/haskell-effectful/effectful/blob/master/transformers.md
+-------------------------------------------------------------------------------
+
+
+{- Note [MonadError on GYTxQueryMonad and ExceptT]
+
+ExceptT instances are omitted since the MonadError requirement for GYTxQueryMonad
+enforces only one exception type: thereby making multiple error handler transformers
+useless.
+See: https://ro-che.info/articles/2014-06-11-problem-with-mtl
+
+Perhaps said superclass constraint should be rethought? Does it really
+achieve what it is meant to achieve? The purpose of that contraint is to signal
+that the query methods _should_ throw errors of type 'GYTxMonadException'.
+After all, some queries are bound to fail. But this expectation does not really
+restrict the type of exceptinos. In fact, an IO based monad could still
+be throwing exceptions of different types (and our implementation of 'GYTxMonadIO'
+likely does).
+
+Alternatively: Use a better effect system. E.g effectful but any ReaderT based effect
+system will suffice (do NOT use free(er) monad like ones). This will trivialize this
+entire problem.
+-}
```

Comment:
Important note on `MonadError` problem. This is essentially the issue that plagues most MTL based systems at one point or another. We can't have multiple `MonadError`s with different errors in the same stack.

The best solution is to really just use an effect system. Not a free(er) monad based one as they are horribly inefficient.